common/source/serial_port.cpp

```diff
@@ -0,0 +1,132 @@
+	parity_			= parity;
+	stop_bits_		= stop_bits;
+	flow_control_	= flow_control;
+	if (ReConfig(config) != kNoError)
+		return make_error_condition(errc::io_error);
+	return kNoError;
+}
+
+error_condition SerialPort::Init(const string& name, const AbstractProtocol& protocol,\
+		int 				baud_rate,\
+		character_size_type character_size,\
+		parity_type 		parity,\
+		stop_bits_type 		stop_bits,\
+		flow_control_type 	flow_control) {
+	CommonKeyPairs config = {
+			{"serial_port_name", name},
+	};
+
+	baud_rate_		= baud_rate;
+	character_size_ = character_size;
+	parity_			= parity;
+	stop_bits_		= stop_bits;
+	flow_control_	= flow_control;
+	if (ReConfig(config) != kNoError)
+		return make_error_condition(errc::io_error);
+	return kNoError;
+}
+
+SerialPort::~SerialPort() {
+
+}
+
+error_condition SerialPort::ReConfig(const CommonKeyPairs &config) {
+	ClearInitialized();
+	auto serial_port_name = config.find(kSerialPortName);
+	if (serial_port_name == config.end())
+		return make_error_condition(errc::invalid_argument);
+
+	serial_port_.open(string(serial_port_name->second.c_str()), ec_);
+	if (ec_)
+		return make_error_condition(errc::bad_address);
+
+	serial_port_.set_option(serial_port::baud_rate(baud_rate_));
+	serial_port_.set_option(serial_port::character_size(character_size_));
+
+	if (parity_ == parity_none)
+		serial_port_.set_option(serial_port::parity(serial_port::parity::none));
+	else if (parity_ == parity_even)
+		serial_port_.set_option(serial_port::parity(serial_port::parity::even));
+	else
+		serial_port_.set_option(serial_port::parity(serial_port::parity::odd));
+
+	if (stop_bits_ == stop_bits_one)
+		serial_port_.set_option(serial_port::stop_bits(serial_port::stop_bits::one));
+	else if (stop_bits_ == stop_bits_onepointfive)
+		serial_port_.set_option(serial_port::stop_bits(serial_port::stop_bits::onepointfive));
+	else
+		serial_port_.set_option(serial_port::stop_bits(serial_port::stop_bits::two));
+
+
+	if (flow_control_ == flow_control_none)
+		serial_port_.set_option(serial_port::flow_control(serial_port::flow_control::none));
+	else if (flow_control_ == flow_control_software)
+		serial_port_.set_option(serial_port::flow_control(serial_port::flow_control::software));
+	else
+		serial_port_.set_option(serial_port::flow_control(serial_port::flow_control::hardware));
+	SetInitialized();
+	return kNoError;
+}
+
+error_condition SerialPort::DeConfig() {
+	serial_port_.close();
+	return kNoError;
+}
+
+error_condition SerialPort::Set(const string& key, const unsigned int value) {
+
+	return kNoError;
+}
+
+error_condition SerialPort::Set(const string& key, const vector<uint8_t>& value) {
+	if (!IsInitialized())
+		return make_error_condition(errc::no_such_device);
+
+	if (key == kSerialPortData) {
+		if (serial_port_.write_some(buffer(value), ec_) == 0)
+			return make_error_condition(errc::io_error);
+	}
+	return kNoError;
+}
+
+error_condition SerialPort::Get(const string& key, unsigned int& result) {
+	if (!IsInitialized())
```

Comment:
Also we get information like `SerialPort::Get("baudrate". baudrate)` in other object which is referencing this class.